app/controllers/api/v0/api_controller.rb

```diff
@@ -5,24 +5,6 @@ class Api::V0::ApiController < ApplicationController
 
   respond_to :json
 
-  def cors_set_access_control_headers
-    headers["Access-Control-Allow-Origin"] = "*"
-    headers["Access-Control-Allow-Methods"] = "POST, GET, PUT, DELETE, OPTIONS"
-    headers["Access-Control-Allow-Headers"] = "Origin, Content-Type, Accept, Authorization, Token"
-    headers["Access-Control-Max-Age"] = "1728000"
```

Comment:
there are a few problematic things with the current configuration:

- origin as `*` though handy is an indication of openness, we'd better echo back the client's origin in the request
- we allow all methods indiscriminately 
- max age doesn't go past 24 hours in Firefox and 2 hours in Chrome anyway https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Max-Age